src/app/api/api.py

```diff
@@ -21,30 +26,87 @@
     field_names_used_for_score_calculation,
 )
 
+
+def get_database(request: Request) -> Database:
+    return request.app.state._db
+
+
 router = APIRouter()
 
+QUALITY_MATRIX_DESCRIPTION = """Calculation of the quality matrix.
+    For each replication source and each property, e.g., `cm:creator`, the quality matrix returns the ratio of
+    elements which miss this entry compared to the total number of entries.
+    A missing entry may be `cm:creator = null`.
+    Additional parameters:
+        store_to_db: Default False. Causes returned quality matrix to also be stored in the backend database."""
+TAG_STATISTICS = "Statistics"
+
 
 @router.get(
     "/quality_matrix",
     status_code=HTTP_200_OK,
     response_model=List[ColumnOutputModel],
     responses={HTTP_404_NOT_FOUND: {"description": "Quality matrix not determinable"}},
-    tags=["Statistics"],
-    description="""Calculation of the quality matrix.
-    For each replication source and each property, e.g., `cm:creator`, the quality matrix returns the ratio of
-    elements which miss this entry compared to the total number of entries.
-    A missing entry may be `cm:creator = null`.""",
+    tags=[TAG_STATISTICS],
+    description=QUALITY_MATRIX_DESCRIPTION,
+)
+async def get_quality_matrix(
+    database: Database = Depends(get_database), store_to_db=False
+):
+    _quality_matrix = await quality_matrix()
+    if store_to_db:
+        await stored_in_timeline(_quality_matrix, database)
+    return _quality_matrix
+
+
+@router.get(
+    "/quality_matrix/{timestamp}",
+    status_code=HTTP_200_OK,
+    response_model=List[ColumnOutputModel],
+    responses={HTTP_404_NOT_FOUND: {"description": "Quality matrix not determinable"}},
+    tags=[TAG_STATISTICS],
+    description=QUALITY_MATRIX_DESCRIPTION
+    + """A timestamp of the format XYZ yields the quality matrix at the respective date.""",
+)
+async def get_past_quality_matrix(
+    timestamp: Optional[int] = None, database: Database = Depends(get_database)
```

Comment:
Why having it optional? If one leaves it out, one gets the plain `/quality_matrix` endpoint. I would return a 400 if the timestamp is missing, or not a valid timestamp. This leads to the next: What is a valid timestamp, is it a unix timestamp in miliseconds, microseconds, seconds? what happens to the time part (hh, mm, ss) of the timestamp? will it be ignored, rounded down/up to the next day?

Reply:
Good point - it is now mandatory. I added additional description of the required timestamp. If the timestamp does not match 1:1 the value in the postgreSQL, nothing is returned.
Would you prefer some more validation of the timestamp, beyond pydantic?